pymodm/fields.py

```diff
@@ -283,10 +283,7 @@ def __init__(self, verbose_name=None, mongo_name=None,
 
         def validate_min_and_max(value):
             # Turn value into a Decimal.
-            if isinstance(value, Decimal128):
-                value = value.to_decimal()
-            elif not isinstance(value, decimal.Decimal):
-                value = decimal.Decimal(value)
+            value = value.to_decimal()
```

Comment:
Note to confused future self: the rationale behind the old pattern (to check the instance type before converting the value to `bson.Decimal128`) was that the field value could be set by the user to be a Python decimal and therefore the validation needed to be able to handle either type because `to_python` wasn't necessarily being called on the field prior to validation. 
With this PR, we are re-introducing the call to `to_python` before validation is performed, thereby obviating the need for the previous type-check.